src/core/control/Control.cpp

```diff
@@ -537,6 +538,12 @@ void Control::actionPerformed(ActionType type, ActionGroup group, GdkEvent* even
         case ACTION_PAPER_BACKGROUND_COLOR:
             changePageBackgroundColor();
             break;
+        case ACTION_MOVE_SELECTION_LAYER_UP:
+            moveSelectionToLayer(getCurrentPage()->getSelectedLayerId());
+            break;
+        case ACTION_MOVE_SELECTION_LAYER_DOWN:
+            moveSelectionToLayer(getCurrentPage()->getSelectedLayerId() - 2);
```

Comment:
The layer id is unsigned, so the subtraction here can cause integer overflow. We should check that `getSelectedLayerId() >= 2` (e.g., ensure that the lowest possible layer number calculated here is 0).

For readability, I would also explicitly indicate that the parameter is a layer number by moving the calculation into another variable:
```suggestion
            Layer::Index currentLayerId = getCurrentPage()->getSelectedLayerId();
            if (currentLayerId >= 2) {
                size_t downLayerNo = currentLayerId - 2;
                moveSelectionToLayer(downLayerNo);
            }
```